### PR TITLE
[MSPAINT] Improve CMiniatureWindow

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fix the miniature window.
JIRA issue: [CORE-18867](https://jira.reactos.org/browse/CORE-18867)

## Proposed changes

- Save the position and size of the miniature window.
- Improve drawing of the miniature window.
- Sync with the canvas.

## Screenshot

image

## TODO

- [x] Do tests.

### CODE REVIEW

File: base/applications/mspaint/miniature.cpp [hunk 1]
```diff
@@ -32,23 +33,72 @@ HWND CMiniatureWindow::DoCreate(HWND hwndParent)
...
+    // Start painting
     PAINTSTRUCT ps;
     HDC hDC = BeginPaint(&ps);
-    StretchBlt(hDC, 0, 0, rc.right, rc.bottom,
-               imageModel.GetDC(), 0, 0, imageModel.GetWidth(), imageModel.GetHeight(),
-               SRCCOPY);
+    if (!hDC)
+        return 0;
+
+    // Use a memory bitmap to reduce flickering
```
Comment (HBelusca) [inline base/applications/mspaint/miniature.cpp:79]: Does this indeed give a visual difference when run on Windows?

File: base/applications/mspaint/history.cpp [hunk 2]
```diff
@@ -17,6 +17,8 @@ void ImageModel::NotifyImageChanged()
 {
     if (canvasWindow.IsWindow())
         canvasWindow.Invalidate(FALSE);
+    if (miniature.IsWindow())
+        miniature.Invalidate(TRUE);
```
Comment (HBelusca) [inline base/applications/mspaint/history.cpp:21]: Is the "TRUE" parameter here used to make the miniature window redraw completely?

Comment (katahiromz) [reply base/applications/mspaint/history.cpp:21]: This `TRUE` is for `::InvalidateRect` function's 3rd parameter `bErase`.
`FALSE` is also OK for this implementation.

File: base/applications/mspaint/miniature.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/miniature.cpp:79]: Yep. The miniature window does flickering.

File: base/applications/mspaint/history.cpp (hunk 2 above)
Comment (HBelusca) [reply base/applications/mspaint/history.cpp]: ok yes, I wanted to understand why the canvas got FALSE while the miniature got TRUE.

Comment (katahiromz) [reply base/applications/mspaint/history.cpp]: Fixed.

**Pull request merged**